### ISSUE TITLE
Different results for validate(), assert() and check()

### ISSUE THREAD

**Issue by micrarior** [Respect/Validation#235]
Am I doing something wrong? 
Why assert and check don't throw exception?
```php
use Respect\Validation\Validator as v;

$array = array('foo' => '');
var_dump(v::key('foo', v::positive())->validate($array)); // false
var_dump(v::key('foo', v::positive())->assert($array)); // true
var_dump(v::key('foo', v::positive())->check($array)); // true
```

**Comment by henriquemoody** [comment]
Actually the but, as I see is because you should have the same result for `validate()`, `assert()` and `check()`, but all values are optional, take a look in the documentation: https://github.com/Respect/Validation#input-optional

**Comment by micrarior** [comment]
> https://github.com/Respect/Validation#input-optional

Sorry, I din't notice this section.
But that means the bug is in the validate method

**Comment by henriquemoody** [comment]
You're right, it is.

**Comment by bmorg** [comment]
The same bug can be reproduced for other validators, e.g. the `numeric` validator:

```php
$validator = v::numeric();
$val = '';

var_dump($validator->validate($val)); // false
var_dump($validator->assert($val)); // true
var_dump($validator->check($val)); // true
```

I am not sure if this is tested for yet.

**Comment by henriquemoody** [comment]
The fix for `key()`, `call()` and `attribute()` rules are available on `0.7.5` and `0.8.1` versions. We still need to fix the behavior on other rules.

**Comment by henriquemoody** [comment]
Will be fixed on version 1.0, see [changelog](https://github.com/Respect/Validation/blob/311891b/CHANGELOG.md#100---2015-10-24) for more info about the version.

To fix this issue now all rules are mandatory, but you can use [optional()](https://github.com/Respect/Validation/blob/311891b/docs/Optional.md) rule to handle optional values, which now is `null` and an empty string by default.
